pid-plugin: Add wrappers for sched_* functions

rohgarg opened the pull request:
Fixes #199.

karya0 commented on src/plugin/pid/sched_wrappers.cpp:
```diff
@@ -0,0 +1,157 @@
+#define _real_sched_setattr         NEXT_FNC(sched_setattr)
+#define _real_sched_getattr         NEXT_FNC(sched_getattr)
+#define _real_sched_setparam        NEXT_FNC(sched_setparam)
+#define _real_sched_getparam        NEXT_FNC(sched_getparam)
```
Can we instead follow the pattern from pidwrappers.h/pid_syscallsreal.c? This plugin is weird in that sense. Until we have a chance to do some cleanup, let's try to keep it consistent! :-)

karya0 commented:
LGTM modulo the _real_ stuff :)

rohgarg commented:
@karya0, @gc00 : Could you please re-review the final patch? I believe it's ready to be merged.

gc00 commented:
It looks good to me.

For my own curiosity, @rohgarg, do you know why the Travis system was choking on compiling your earlier implementation of the `_real_` stuff?  Maybe there are some lessons for me, too.  :-)

rohgarg commented:
@gc00: There were two separate issues:
 - There are two functions -- `sched_getattr()` and `sched_setattr()` -- that the man page says are present since kernel ver. 3.14. The man page also seems to imply that the function declarations and a `struct sched_attr` that these functions reference are present in a libc public header file: `sched.h`. And so, I added wrappers for these two, but turns out that neither of the two things I mentioned before are  exported publicly by libc.

  My fix is to ifdef out the two function wrappers; we can enable them in the future, if there is a requirement.

 - The second problem was with the test that I had added to the make-check test suite. The container environment Travis runs, apparently, doesn't support calls to setaffinity. My test, however, would assert and exit when it couldn't change the affinity of a running process. Note that I this wasn't a problem on other systems I tested.

  My fix is to comment out the asserts in the test program.

karya0 commented on src/plugin/pid/pid_syscallsreal.c:
```diff
@@ -411,3 +412,52 @@ ssize_t _real_readlink(const char *path, char *buf, size_t bufsiz) {
+  REAL_FUNC_PASSTHROUGH(sched_getparam) (pid, param);
+}
+
+#if 0
```
Can we instead add a configure check for these functions?
